jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/util/PersistenceUtil.java

```diff
@@ -71,4 +221,127 @@ public static StatelessSession openStatelessSession(EntityManagerFactory entityM
 		return statelessSession;
 	}
 
+	/**
+	 * Creates a list of {@link Order} based on the ID attribute(s) inside the
+	 * given entity type.
+	 *
+	 * @param entityManagerFactory Entity manager factory.
+	 * @param entity The entity type.
+	 * @param <X> The type containing the represented attribute.
+	 *
+	 * @return A list of {@link Order}, sorted by lexicographical order on the
+	 * ID attributes' name.
+	 */
+	public static <X> List<Order> createIdOrders(EntityManagerFactory entityManagerFactory, Class<X> entity) {
+		EntityType<X> entityType = entityManagerFactory.getMetamodel().entity( entity );
+		List<SingularAttribute<?, ?>> attributeList;
+		List<Order> orders = new ArrayList<>();
+
+		switch ( getIdTypeOf( entityType ) ) {
+			case SINGLE_ID:
+				Class<?> idJavaType = entityType.getIdType().getJavaType();
+				SingularAttribute<?, ?> idAttribute = entityType.getId( idJavaType );
+				orders.add( Order.asc( idAttribute.getName() ) );
+				return orders;
+
+			case EMBEDDED_ID:
+				Class<?> embeddable = entityType.getIdType().getJavaType();
+				EmbeddableType<?> embeddableType = entityManagerFactory.getMetamodel().embeddable( embeddable );
+				String embeddableName = entityType.getId( embeddable ).getName();
+
+				attributeList = new ArrayList<>( embeddableType.getSingularAttributes() );
+				attributeList.sort( Comparator.comparing( Attribute::getName ) );
+				attributeList.forEach( attr -> {
+					String propertyName = embeddableName + "." + attr.getName();
+					orders.add( Order.asc( propertyName ) );
+				} );
+				return orders;
+
+			case ID_CLASS:
+				attributeList = new ArrayList<>( entityType.getIdClassAttributes() );
+				attributeList.sort( Comparator.comparing( Attribute::getName ) );
+				attributeList.forEach( attr -> orders.add( Order.asc( attr.getName() ) ) );
+				return orders;
+
+			default:
+				throw new IllegalStateException( "Cannot determine IdType: this should never happen." );
```

Comment:
You're right. By doing using the `org.hibernate.search.exception.AssertionFailure`, we clarify the responsibility of the exception: it is an internal error, and users should report a bug if they meet it. This is also written in the Javadoc of `AssertionFailure`:

> Indicates failure of an assertion: a possible bug in Hibernate Search.